File: app/src/main/java/org/dicio/dicio_android/output/speech/InstantSpeechDevice.java
package org.dicio.dicio_android.output.speech;

import org.dicio.skill.output.SpeechOutputDevice;

/**
 * A speech device that speaks instantaneously (e.g. because it just displays text). Overrides
 * {@link #isSpeaking()} so that it always returns false, and {@link
 * #runWhenFinishedSpeaking(Runnable)} so that the runnable is run instantly.
 */
public abstract class InstantSpeechDevice implements SpeechOutputDevice {
    @Override
    public final boolean isSpeaking() {
        return false;
    }

    @Override
    public final void runWhenFinishedSpeaking(final Runnable runnable) {
        runnable.run();
    }
}

File: app/src/main/kotlin/org/stypox/dicio/output/speech/InstantSpeechDevice.kt
package org.stypox.dicio.output.speech

import org.dicio.skill.output.SpeechOutputDevice

/**
 * A speech device that speaks instantaneously (e.g. because it just displays text). Overrides
 * [isSpeaking] so that it always returns false, and [runWhenFinishedSpeaking] so that the runnable is run instantly.
 */
abstract class InstantSpeechDevice : SpeechOutputDevice {
    override fun isSpeaking(): Boolean {
        return false
    }

    override fun runWhenFinishedSpeaking(runnable: Runnable) {
        runnable.run()
    }
}